sections/exchange/hooks/useExchange.tsx

```diff
@@ -602,28 +602,82 @@ const useExchange = ({
 		oneInchQuoteQuery.isSuccess,
 	]);
 
-	const getExchangeParams = useCallback(() => {
-		const quoteKeyBytes32 = ethers.utils.formatBytes32String(quoteCurrencyKey!);
-		const baseKeyBytes32 = ethers.utils.formatBytes32String(baseCurrencyKey!);
-		const amountToExchange = quoteCurrencyAmountBN.toBN();
-		const trackingCode = ethers.utils.formatBytes32String('KWENTA');
+	const getExchangeParams = useCallback(
+		(isAtomic?: boolean) => {
+			if (isAtomic === true) {
+				const destinationCurrencyKey = ethers.utils.formatBytes32String(quoteCurrencyKey!);
+				const sourceCurrencyKey = ethers.utils.formatBytes32String(baseCurrencyKey!);
+				const sourceAmount = quoteCurrencyAmountBN.toBN();
+				const trackingCode = ethers.utils.formatBytes32String('KWENTA');
+
+				return [destinationCurrencyKey, sourceAmount, sourceCurrencyKey, trackingCode];
+			} else {
+				const quoteKeyBytes32 = ethers.utils.formatBytes32String(quoteCurrencyKey!);
+				const baseKeyBytes32 = ethers.utils.formatBytes32String(baseCurrencyKey!);
+				const amountToExchange = quoteCurrencyAmountBN.toBN();
+				const trackingCode = ethers.utils.formatBytes32String('KWENTA');
 
-		return [quoteKeyBytes32, amountToExchange, baseKeyBytes32, walletAddress, trackingCode];
-	}, [baseCurrencyKey, quoteCurrencyAmountBN, quoteCurrencyKey, walletAddress]);
+				return [quoteKeyBytes32, amountToExchange, baseKeyBytes32, walletAddress, trackingCode];
+			}
+		},
+		[baseCurrencyKey, quoteCurrencyAmountBN, quoteCurrencyKey, walletAddress]
+	);
 
 	const getGasEstimateForExchange = useCallback(
 		async (gasPriceInWei: number | null) => {
 			try {
 				if (isL2 && !gasPrice) return null;
 				if (synthetixjs != null) {
-					const exchangeParams = getExchangeParams();
-					const gasEstimate = await synthetixjs.contracts.Synthetix.estimateGas.exchangeWithTracking(
-						...exchangeParams
-					);
-					let gasLimitNum = Number(gasEstimate);
-					const metaTx = await synthetixjs.contracts.Synthetix.populateTransaction.exchangeWithTracking(
-						...exchangeParams
-					);
+					const destinationCurrencyKey = getExchangeParams()[0];
+
+					let gasEstimate, gasLimitNum, metaTx, isAtomic;
+
+					if (
+						destinationCurrencyKey === 'sBTC' ||
+						destinationCurrencyKey === 'sETH' ||
+						destinationCurrencyKey === 'sEUR'
+					) {
+						isAtomic = true;
+
+						const exchangeParams = getExchangeParams(isAtomic);
```

Comment:
A cleaner more [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) way to do this would be:

```
const isAtomic = destinationCurrencyKey === 'sBTC' ||
			 destinationCurrencyKey === 'sETH' ||
			 destinationCurrencyKey === 'sEUR';
const exchangeParams = getExchangeParams(isAtomic);

// then your if (isAtomic) {...} blocks
```

Reply:
You can still make it more DRY by moving 
```
const exchangeParams = getExchangeParams(isAtomic);
```
out of the if blocks. It's repeated quite a few times.